process/smartContract/scQueryService.go

```diff
@@ -193,15 +195,14 @@ func (service *SCQueryService) executeScCall(query *process.SCQuery, gasPrice ui
 	}
 
 	if len(blockRootHash) > 0 {
+		logQueryService.Trace("preparing execution for block and root hash", "block", blockHeader.GetNonce(), "rootHash", blockRootHash)
```

Comment:
the check for nil from L243 should also be done before this log

also needed for recreateTrie

Reply:
Fixed (moved log in `recreateTrie`, as well). Could not reproduce this scenario using a localnet. VM queries are not available at genesis, though.